### ISSUE TITLE
Rule Change: Incorrect error message for CSS functions in `use-baseline` rule

### ISSUE THREAD

**Issue by thecalamiity** [eslint/css#293]
### What rule do you want to change?

use-baseline

### What change do you want to make?

Other

### How do you think the change should be implemented?

Other

### Example code

```css
.a {
  color: light-dark(#333b3c, #efefec);
}
```

### What does the rule currently do for this code?

The rule reports:

```
Type 'light-dark' is not a widely available baseline feature.
```

This is confusing because `light-dark` is clearly a CSS function, not a "type". The term "type" is misleading and doesn't accurately describe what's being checked.

### What will the rule do after it's changed?

After the change, the rule will report:

```
Function 'light-dark' is not a widely available baseline feature.
```

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

The data generation script already correctly filters to only include CSS functions (by checking `mdnData.css.functions`), but currently exports them under the misleading name `types`. We should update the data export name to `functions` for consistency.
